### PR TITLE
z_bg_jya_haheniron OK

### PR DESCRIPTION
**Pull request by jpburnett**

### CODE REVIEW

File: src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c [hunk 1]
```diff
@@ -29,27 +34,162 @@ const ActorInit Bg_Jya_Haheniron_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Haheniron/BgJyaHaheniron_Update.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_CONTINUE),  ICHAIN_F32_DIV1000(minVelocityY, -15000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Haheniron/BgJyaHaheniron_Draw.s")
+f32 D_80898794[] = { 0.129999995232f, 0.10000000149f, 0.10000000149f };
```
[the remark this answers is not in the document]
Comment (jpburnett) [reply src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c:59]: It did work! I had tried limiting the significant figures on them, but didn't consider just rounding up the 12.9f to 13.0f

File: src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c [hunk 2]
```diff
@@ -29,27 +34,160 @@ const ActorInit Bg_Jya_Haheniron_InitVars = {
...
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Haheniron/BgJyaHaheniron_Draw.s")
+f32 D_80898794[] = { 0.13f, 0.1f, 0.1f };
+
+Vec3f D_808987A0[] = { 0.0f, 14.0f, 0.0f };
+
+Vec3f D_808987AC[] = { 0.0f, 8.0f, 0.0f };
+
+extern Gfx D_06000880;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c:65]:
```suggestion
extern Gfx D_06000880[];
```

File: src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c [hunk 3]
```diff
@@ -29,27 +34,160 @@ const ActorInit Bg_Jya_Haheniron_InitVars = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Haheniron/func_8089861C.s")
+static ColliderJntSphInit D_80898764 = {
+    { COLTYPE_UNK10, 0x11, 0x01, 0x00, 0x00, COLSHAPE_JNTSPH },
+    1,
+    sJntSphItemsInit,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Haheniron/func_8089862C.s")
+static s16 kakeraScale[] = { 5, 8, 11, 14, 17 };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Jya_Haheniron/z_bg_jya_haheniron.c:51]:
```suggestion
static s16 sKakeraScales[] = { 5, 8, 11, 14, 17 };
```

**Review by Roman971 — COMMENTED** [review]
Just a few remaining things

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Comment by fig02** [comment]
small conflict that needs fixing, i think you just need to merge master and then remove the asm file

**Pull request merged**